### ISSUE TITLE
store server kindof-silently fails when directed at a store at the root of a mounted share drive

### ISSUE THREAD

**Issue by josh-hemphill** [pnpm/pnpm#6386]
### Code to reproduce the issue:

1. Have a windows share and mount it as a drive
2. initialize a store directory at the root of the drive
3. run `pnpm server start` on the drive
4. Attempt to run any pnpm command that uses the store

### Expected behavior:

To operate using the store on the drive

### Actual behavior:

It appears to run until operations are run against it, then throwing tons of warnings:

On the client:
```
WARN POST http://localhost:5813/packageFilesResponse error (503). Will retry in 1 minute. 99 retries left.
```

On the PNPM server terminal: (includes this warning for all kinds of packages, including: `fsevents` and `@esbuild/...`)
```
WARN Unsupported platform for registry.npmjs.org/...
```

### Additional information:

related: #4391

 - `node -v` prints: 18.15.0
 - Windows, macOS, or Linux?: Windows

**Comment by josh-hemphill** [comment]
For us, the network share is *not* being use a store for multiple systems, it only exists because our VM and storage are provisioned separately. We have multiple users all simultaneously on the same system, it just happens the folders where our projects are, have to be on a share drive. So that's why we have to use PNPM store server, it's only being accessed by the users on the one VM on which its mounted.

**Comment by zkochan** [comment]
I am not sure why you need the store server. If it is on a different volume then it will be slow as pnpm will just copy the files. If you want to have some kind of shared cache, I think a better approach would be to run verdaccio instead with the storage located on that other volume (where you run the pnpm store server). Then just run pnpm with the registry set to your local verdaccio.

**Comment by josh-hemphill** [comment]
We don't really care much about the cache itself, we're mostly just trying to reduce storage consumption by combining users pnpm directories wherever it's possible.

**Comment by zkochan** [comment]
I don't think you need a store server for that. You can run install concurrently by many users on the same store, it won't break the store. I don't understand your architecture but if the store on a different filesystem, then you won't reduce consumption as hard links only work inside the same disk.

**Comment by josh-hemphill** [comment]
That's why I was trying to get it to work on share, all the projects and everything is being run on the share, so in theory, with everything local to the share, the links should still work.

**Comment by zkochan** [comment]
ok, so I guess you just need to set the `store-dir` setting to the location where you want the store to be located. All operations on the store are atomic, so it won't be left in a broken state.

**Comment by josh-hemphill** [comment]
I think where I run into trouble is if I set the global `store-dir` to be the one on the share, then if anyone tries to run like a `pnpm i -g` to update something, before they navigate to the share, then somehow their user config changes to use a user-specific store file and we return to having issues.
There was something happening with configs changing back to incorrectly report store files in user directories.

**Comment by josh-hemphill** [comment]
Is there some way to inherit configs based on the current drive?

**Comment by zkochan** [comment]
I don't think so. Currently we use npm's config system. Maybe you can try to set a custom location for the global `.npmrc` file by changing an env variable.

**Comment by josh-hemphill** [comment]
I'm already doing that, but I have it set to the C drive, I guess putting it on the share would make sense

**Comment by josh-hemphill** [comment]
I don't have access to my work setup at the moment to double check what we settled on. But we did just set it to use a directory on the share drive, but we did have to forego using the store server since we couldn't get that working with it, but it seems to be fine using the integrated store, it just means the people we have that occasionally try to use pnpm on the main drive run into issues trying to pull everything over the network when there's already a cache on another drive.

**Comment by zkochan** [comment]
This targets a part of pnpm that no longer exists, so closing.

The store server has been removed from pnpm.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
